src/cpp/src/text2image/stable_diffusion_3_pipeline.hpp

```diff
@@ -0,0 +1,576 @@
+        m_clip_text_encoder_1->reshape(batch_size_multiplier);
+        m_clip_text_encoder_2->reshape(batch_size_multiplier);
+        m_vae_decoder->reshape(num_images_per_prompt, height, width);
+        m_transformer->reshape(num_images_per_prompt * batch_size_multiplier,
+                               height,
+                               width,
+                               m_clip_text_encoder_1->get_config().max_position_embeddings);
+    }
+
+    void compile(const std::string& device, const ov::AnyMap& properties) override {
+        m_clip_text_encoder_1->compile(device, properties);
+        m_clip_text_encoder_2->compile(device, properties);
+        m_vae_decoder->compile(device, properties);
+        m_transformer->compile(device, properties);
+    }
+
+    ov::Tensor generate(const std::string& positive_prompt, const ov::AnyMap& properties) override {
+        using namespace numpy_utils;
+        GenerationConfig generation_config = m_generation_config;
+        generation_config.update_generation_config(properties);
+
+        const auto& transformer_config = m_transformer->get_config();
+        const size_t batch_size_multiplier = do_classifier_free_guidance(generation_config.guidance_scale)
+                                                 ? 2
+                                                 : 1;  // Transformer accepts 2x batch in case of CFG
+
+        size_t vae_out_channels_size = m_vae_decoder->get_config().block_out_channels.size();
+        const size_t vae_scale_factor = std::pow(2, vae_out_channels_size - 1);
+
+        if (generation_config.height < 0)
+            generation_config.height = transformer_config.sample_size * vae_scale_factor;
+        if (generation_config.width < 0)
+            generation_config.width = transformer_config.sample_size * vae_scale_factor;
+
+        check_image_size(generation_config.height, generation_config.width);
+
+        if (generation_config.random_generator == nullptr) {
+            uint32_t seed = time(NULL);
+            generation_config.random_generator = std::make_shared<CppStdGenerator>(seed);
+        }
+
+        // Input tensors for transformer model
+        ov::Tensor prompt_embeds_inp, pooled_prompt_embeds_inp;
+
+        // 1. Encode positive prompt:
+        std::string prompt_2_str =
+            generation_config.prompt_2 != std::nullopt ? *generation_config.prompt_2 : positive_prompt;
+        std::string prompt_3_str =
+            generation_config.prompt_3 != std::nullopt ? *generation_config.prompt_3 : positive_prompt;
+
+        ov::Tensor pooled_prompt_embed_out = m_clip_text_encoder_1->infer(positive_prompt, "", false);
+        size_t idx_hidden_state_1 = m_clip_text_encoder_1->get_config().num_hidden_layers + 1;
+        ov::Tensor prompt_embed_out = m_clip_text_encoder_1->get_output_tensor(idx_hidden_state_1);
+
+        ov::Tensor pooled_prompt_2_embed_out = m_clip_text_encoder_2->infer(prompt_2_str, "", false);
```

Comment:
why don't you infer both negative and positive prompts in a single call like 
```suggestion
        ov::Tensor pooled_prompt_2_embed_out = m_clip_text_encoder_2->infer(prompt_2_str, negative_prompt_2_str, do_classifier_free_guidance(generation_config.guidance_scale));
```